### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by leishman** [ElementsProject/lightning#1593]
### Issue and Steps to Reproduce
When I run `make` after running `./configure` in the root directory I get the following error:

```
backtrace -L/usr/local/lib -lm -lgmp -lsqlite3 -lz  -o ccan/ccan/cdump/tools/cdump-enumstr
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:326: recipe for target 'ccan/ccan/cdump/tools/cdump-enumstr' failed
make: *** [ccan/ccan/cdump/tools/cdump-enumstr] Error 1
```

Looks like my machine is now missing some dependency but it's unclear to me what that is.

### `getinfo` output

This error is on the head of master and it didn't build so this command wouldn't output the relevant information.

**Comment by JohnnyBackflip** [comment]
This fixed it for me:

sudo apt-get install libgtk2.0-dev intltool editorconfig libeditorconfig-dev autoconf automake build-essential libtool libgmp-dev python python3 net-tools libsodium-dev

But I don't know which of the dependency exactly fixed the error

**Comment by deinerson1** [comment]
Installing zlib1g-dev fixed this on Ubuntu amd64 18.04 LTS bionic

`sudo apt-get install zlib1g-dev`

**Comment by cdecker** [comment]
Yes, the `zlib` dependency was recently introduced for the selective gossip syncing.
